File: output/copilot/java/Solution13_cp.java
public class Solution13_cp {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = { -1, -1 };
        int n = nums.length;
        int l = 0;
        int r = n - 1;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (nums[mid] == target) {
                int i = mid;
                int j = mid;
                while (i >= 0 && nums[i] == target) {
                    i--;
                }
                while (j < n && nums[j] == target) {
                    j++;
                }
                ans[0] = i + 1;
                ans[1] = j - 1;
                return ans;
            } else if (nums[mid] < target) {
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }
        return ans;
    }

}

File: output/copilot/js/Solution13_cp.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var searchRange = function(nums, target) {
    let start = 0;
    let end = nums.length - 1;
    let mid = Math.floor((start + end) / 2);
    let result = [-1, -1];
    while (start <= end) {
        if (nums[mid] === target) {
            result[0] = mid;
            result[1] = mid;
            let left = mid - 1;
            let right = mid + 1;
            while (nums[left] === target) {
                result[0] = left;
                left--;
            }
            while (nums[right] === target) {
                result[1] = right;
                right++;
            }
            return result;
        } else if (nums[mid] < target) {
            start = mid + 1;
        } else {
            end = mid - 1;
        }
        mid = Math.floor((start + end) / 2);
    }
    return result;
    
};